saltswap/wrappers.py

```diff
@@ -378,6 +382,164 @@ def update(self, nattempts=None, chemical_potential=None, saltmax=None):
+    converge if the initial bias is far from the negative of the true free energy of the salt numbers. SAMSalinator
+    can be supplied with initial values of the biases via the 'initial_bias' flag. It's use is strongly recommended for
+    production simulations.
+
+    Initialize a netcdf file to store the simulation data, making sure to add the special commands to record the SAMS
+    weights.
+    >>> creator = Record.CreateNetCDF('out.nc')
+    >>> ncfile = creator.create_netcdf(sams_salinator.swapper, nstates=nstates)
+
+    Runing a short simulation by mixing MD and MCMC saltswap moves and saving the simulation data.
+    >>> for i in range(100):
+    >>> ...     langevin.step(2000)
+    >>> ...     sams_salinator.update()
+    >>> ...     Record.record_netcdf('out.nc', context, sams_salinator.swapper, i,sams_bias=sams_salinator.bias)
+
+    All the data required to calibrate the chemical potential can be found in 'out.nc'.
+    """
+    def __init__(self, saltmin=0, saltmax=20, initial_bias=None, two_stage=True, beta=0.7, target_weights=None,
+                 precision=0.1, **kwargs):
+        """
+        Parameters
+        ----------
+        saltmin: int
+            the minimum number of salt pairs that will be sampled over.
+        saltmax: int
+            the maximum number of salt pairs that will be sampled over.
+        initial_bias: numpy.ndarray
+            array of the initial values of the SAMS biases.
+        two_stage: bool
+            whether to perform a two-stage procedure, where the first is a burn-in.
+        beta: float
+            the exponent of the gain in the two-stage procedure. Should be between 0.5 and 1.
+        target_weights: numpy.ndarray
+            the target state sampling proportions that the biases will be optimized to obtain.
+        precision: float
+            the precision to which the sampling proportions will match the target weights before the two-stage
+            procedure is terminated.
+
+        """
+        if saltmax <= saltmin:
+            raise Exception('The maximum amount of salt must be less than the minimum.')
+
+        super(SAMSSalinator, self).__init__(**kwargs)
+
+        self.saltmin = saltmin
+        self.saltmax = saltmax
+        self.nstates = 1 + saltmax - saltmin
+        self.state_counts = np.zeros(self.nstates)
+        self.nsalt = self.count_salt()
+
+        if initial_bias is None:
+            self.bias = np.zeros(self.nstates)
+        else:
+            self.bias = initial_bias
+
+        self.adaptor = SAMSAdaptor(self.nstates, zetas=self.bias, target_weights=target_weights, two_stage=two_stage,
+                                   beta=beta, precision=precision)
+    def count_salt(self):
+        """
+        Count and return the number of neutral anion and cation pairs.
+        """
+        nwater, ncation, nanion = self.swapper.get_identity_counts()
```

Comment:
The class `SAMSSalinator` inherits a lot from the `Salinator` class. I'll add a comment here to make this clearer.